WIP mark archives generated by gym with extended attribute

mpirri opened the pull request:
NOTE This PR is on my own fork of gym because I do not wish to have discussion of this in the public fastlane repos (since this change is Fabric-internal motivated).  Hemal agreed with the decision to do it this way.

This is a change to mark archives generated by gym to allow them to be differentiated from archives generated by Xcode. This ability is required by the Mac app (changes are in a separate PR) to allow it to disable the "distribute this archive with Crashlytics Beta" user notification & prompt, which currently fires after gym finishes building any archive, and is confusing for users.

Biggest concern with this change comes with the fact that we are adding functionality to fastlane/gym to support Fabric mac app. Although it is not impossible to imagine a situation where this might be useful to another tool, it is most likely to only ever be used by Fabric.

How do we feel about this? Modifying our open-source project with code to support closed-source Fabric functionality? Thoughts?

ohwutup commented:
This seems pretty chill to me. I think that it could definitely be useful to have this flag for other purposes beyond ours. :rocket:

KrauseFx commented on lib/gym/runner.rb:
```diff
@@ -71,6 +71,10 @@ def fix_package
     end
 
+    def mark_archive_as_built_by_gym(archive_path)
+      system "xattr -w info.fastlane.generated_by_gym 1 \"#{archive_path}\""
```
Please surround with `(`

KrauseFx replied:
The `"` archive_path will have issues if there is a `"` in the file name. Maybe we should use the `shellescape` method instead.

KrauseFx commented:
Great choice on doing a private fork on this - great idea using the `xcattr` method to do so. The only thing that we need to check first, is that this change doesn't conflict with the iTunes Connect upload and app submission. In particular I wonder if this change somehow breaks the code signing attributes of the binary. Probably not, as it only affects the archive, but we still need to test this :+1:

KrauseFx commented:
Uploading a binary to iTunes Connect and see if it can be processed should be enough to test this.

mpirri commented:
@KrauseFx I checked uploading to iTunes Connect and then addressed your PR feedback and opened a PR on the fastlane/gym repository: https://github.com/fastlane/gym/pull/206
Let me know what you think!

@ohwutup here is the associated Mac app change to look at the xattr and the disable the beta prompt if it is set: https://github.com/crashlytics/apple/pull/4322
